src/main/java/org/cbioportal/service/impl/StudyViewColumnarServiceImpl.java

```diff
@@ -97,14 +97,60 @@ public Map<String, ClinicalDataType> getClinicalAttributeDatatypeMap() {
     
     @Override
     public List<ClinicalDataCountItem> getClinicalDataCounts(StudyViewFilter studyViewFilter, List<String> filteredAttributes) {
-        return studyViewRepository.getClinicalDataCounts(createContext(studyViewFilter), filteredAttributes)
+        Map<String, ClinicalDataType> clinicalAttributeDatatypeMap = this.getClinicalAttributeDatatypeMap();
+        StudyViewFilterContext studyViewFilterContext = createContext(studyViewFilter);
+            
+        int filteredSamplesCount = studyViewRepository.getFilteredSamplesCount(studyViewFilterContext);
```

Comment:
unless i'm mistaken, we are executing all the filtering logic three times in a row here with each of these calls to the repository layer?

Reply:
@onursumer is this true?

Reply:
I think it is true. Not sure if there is a way to avoid it.

Reply:
lets keep keep it the way it is and revisit if performance needs dictate